src/test/java/org/jenkinsci/plugins/docker/commons/credentials/DockerRegistryEndpointTest.java

```diff
@@ -111,15 +121,27 @@ public void testGetTokenForRun() throws IOException {
                 globalCredentialsId, "test-global-creds", "user", "password");
         CredentialsProvider.lookupStores(j.jenkins).iterator().next().addCredentials(Domain.global(), credentials);
 
-        FreeStyleProject item = j.createFreeStyleProject();
+        FreeStyleProject p1 = j.createFreeStyleProject();
+        FreeStyleProject p2 = j.createFreeStyleProject();
+
+        Map<String, Authentication> jobsToAuths = new HashMap<>();
+        jobsToAuths.put(p1.getName(), User.getById("alice", true).impersonate());
+        jobsToAuths.put(p2.getName(), User.getById("bob", true).impersonate());
+        QueueItemAuthenticatorConfiguration.get().getAuthenticators().replace(new MockQueueItemAuthenticator(jobsToAuths));
 
         try (ACLContext as = ACL.as(User.getById("alice", false))) {
-            DockerRegistryToken token = new DockerRegistryEndpoint("https://index.docker.io/v1/",
-                    globalCredentialsId).getToken(new FreeStyleBuild(item));
-            Assert.assertNotNull(token);
+            DockerRegistryToken token = new DockerRegistryEndpoint("https://index.docker.io/v1/", globalCredentialsId)
+                    .getToken(j.buildAndAssertSuccess(p1));
+            Assert.assertNotNull("Alice has Credentials.USE_ITEM and should be able to use the credential", token);
             Assert.assertEquals("user", token.getEmail());
             Assert.assertEquals(Base64.getEncoder().encodeToString("user:password".getBytes(Charsets.UTF_8)), token.getToken());
         }
+
+        try (ACLContext as = ACL.as(User.getById("bob", false))) {
+            DockerRegistryToken token = new DockerRegistryEndpoint("https://index.docker.io/v1/", globalCredentialsId)
+                    .getToken(j.buildAndAssertSuccess(p2));
```

Comment:
Not obvious from reading this that the authentication is being applied to `getToken` and ignored by `buildAndAssertSuccess`. Would be clearer to extract the `Run` variable outside the `as` block.

Reply:
Good point, done in 65ea219.